tests/results.cpp

```diff
@@ -2027,3 +2027,184 @@ TEST_CASE("distinct") {
+        }
+
+        SECTION("min") {
+            Results results(r, table->where());
+
+            REQUIRE(results.min(0)->get_int() == 0);
+            REQUIRE(results.min(1)->get_float() == 0.f);
+            REQUIRE(results.min(2)->get_double() == 0.0);
+            REQUIRE(results.min(3)->get_timestamp() == Timestamp(0, 0));
+        }
+
+        SECTION("average") {
+            Results results(r, table->where());
+
+            REQUIRE(results.average(0)->get_double() == 1.0);
+            REQUIRE(results.average(1)->get_double() == 1.0);
+            REQUIRE(results.average(2)->get_double() == 1.0);
+            REQUIRE_THROWS_AS(results.average(3), Results::UnsupportedColumnTypeException);
+        }
+
+        SECTION("sum") {
+            Results results(r, table->where());
+
+            REQUIRE(results.sum(0)->get_int() == 2);
+            REQUIRE(results.sum(1)->get_double() == 2.0);
+            REQUIRE(results.sum(2)->get_double() == 2.0);
+            REQUIRE_THROWS_AS(results.sum(3), Results::UnsupportedColumnTypeException);
+        }
+    }
+
+    SECTION("rows with all null values") {
+        const int row_count = 3;
+        r->begin_transaction();
+        table->add_empty_row(row_count);
+        for (int i = 0; i < column_count; ++i) {
+            for (int j = 0; j < row_count; ++j) {
+                table->set_null(i, j);
+            }
+        }
+        // table:
+        //  null, null, null,  null,  null
+        //  null, null, null,  null,  null
+        //  null, null, null,  null,  null
+        r->commit_transaction();
+
+        SECTION("max") {
+            Results results;
+
+            SECTION("results built from query") {
+                results = Results(r, table->where());
+            }
+
+            REQUIRE(!results.max(0));
+            REQUIRE(!results.max(1));
+            REQUIRE(!results.max(2));
+            REQUIRE(!results.max(3));
+        }
+
+        SECTION("min") {
+            Results results(r, table->where());
+
+            REQUIRE(!results.min(0));
+            REQUIRE(!results.min(1));
+            REQUIRE(!results.min(2));
+            REQUIRE(!results.min(3));
+        }
+
+        SECTION("average") {
+            Results results(r, table->where());
+
+            REQUIRE(results.average(0)->get_double() == 0.0);
+            REQUIRE(results.average(1)->get_double() == 0.0);
+            REQUIRE(results.average(2)->get_double() == 0.0);
+            REQUIRE_THROWS_AS(results.average(3), Results::UnsupportedColumnTypeException);
+        }
+
+        SECTION("sum") {
+            Results results(r, table->where());
+
+            REQUIRE(results.sum(0)->get_int() == 0);
+            REQUIRE(results.sum(1)->get_double() == 0.0);
+            REQUIRE(results.sum(2)->get_double() == 0.0);
+            REQUIRE_THROWS_AS(results.sum(3), Results::UnsupportedColumnTypeException);
+        }
+    }
+
+    SECTION("empty") {
```

Comment:
It'd be worth distinguishing between `Mode::Empty`, an empty table, and a query that matched no rows. They each take slightly different paths within `Results::aggregate`.